bfasst/utils/structural.py

```diff
@@ -693,11 +707,14 @@ def check_for_potential_bram_mapping(self, named_instance):
                     raise StructuralCompareError("Unexpected BRAM CASCADE Configuration")
                 logging.info("Unexpected BRAM CASCADE Configuration for %s", named_instance.name)
 
-        instances_matching_connections = [
-            i for i in self.possible_matches[named_instance] if i not in self.block_mapping.inverse
-        ]
-
-        for pin in named_instance.pins:
+        instances_matching_connections = self.possible_matches[instance_name] - set(
```

Comment:
Is there a significant cost to having to cast self.block_mapping.inverse to a set?

Reply:
I don't think so, but I haven't been able to find a cast method or __init__ method for set creation in the profile, so I can't actually see. I think we need to break this file into a lot more functions. It would make for a more accurate profile because then we would really be able to tell exactly which calls are taking up the most time.

Reply:
For example, in check_for_potential_mapping and check_for_potential_bram_mapping, we have this redundant code:

`tmp = {
                instance
                for instance in instances_matching_connections
                if self.reversed_instance_map[instance].get_pin(name, idx).net == other_net
            }
        if not tmp:
            if other_net.is_gnd:
                tmp = {
                    instance
                    for instance in instances_matching_connections
                    if self.reversed_instance_map[instance].get_pin(name, idx).net.is_gnd
                }
            elif other_net.is_vdd:
                tmp = {
                    instance
                    for instance in instances_matching_connections
                    if self.reversed_instance_map[instance].get_pin(name, idx).net.is_vdd
                }`